lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -293,10 +287,9 @@ def _on_scrape_targets_changed(self, event):
 (https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config).
 
 Units of consumer charm advertise their address over unit relation
-data using the `prometheus_scrape_host` key. While the
+data using the `prometheus_scrape_host_address` key. While the
```

Comment:
@balbirthomas please remind me why we did this? It's going to come _really_ in handy with `ingress`, but I am not sure we had that in mind at the time.

Reply:
When the Prometheus job specification use wildcard hosts (`*:8080`) then `MetricsEndpointConsumer` needs the IP addresses of all the units. There were two options at that time 1) `MetricsEndpointConsumer` should gather the IP address from relation event data 2) `MetricsEndpointProvider` units forward this data in unit relation data. I am not sure if I remember the precise details why we went down route 2 but I think at that time there was lot of discussion on how this should be done. The consensus in the team was for the provider charm in this case to forward the IP address through unit relation data. If I remember correctly this was the pattern Jeremy was recommending. I do remember creating a [ticket](https://github.com/canonical/operator/issues/560) for `ops` also. I think Graylog charm (at least Jeremy's branch) and other charms may also be doing the same. Happy to investigate if you think this needs changing.